### PR TITLE
Marius_Bauer

### PR DESCRIPTION
**Pull request by Marius-Bauer93**
removing mentioned errors and typos from the code.

### CODE REVIEW

**Comment by Marius-Bauer93** [comment]
@timriker Please let me know which details I need to add to the IProfile.ts file, do not what this is for. Thanks!

**Comment by timriker** [comment]
please rename the .ts file back to a .ts file.

The js-src is where the "source code" to the javascript files lives. It's in TypeScript. The IProfile.ts file should describe the interface to the Profile such that typescript can validate it is being used correctly.

I run:

tsc --build tsconfig.json --verbose

to convert the js-src/*.ts files to underlay/js/*.js files. If I do this on your patch, all your js file edits will be reverted. Copy your edited into js-src/ then flush out the IProfile.ts interface file so syntax checking works. Then run tsc and validate that your changes are still there.

For local testing, I set my ReportRoot to "underlay" so I can test right in my working directory. I point apache at that directory so everything loads correctly.

**Comment by timriker** [comment]
You added a new version in the comments. That's a Good Thing. You should also edit:

$Global:ScriptVersion in bigipreport.ps1
"Settings version" in bigipreportconfig.xml

so that the versions match.

File: js-src/SiteDataInterfaces/IProfile.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export default interface IProfile {
+    //what should be in here?
+}
```
Comment (epacke) [inline js-src/SiteDataInterfaces/IProfile.ts:3]: Missing new line, please add.

File: underlay/js/bigipreport.js [hunk 2]
```diff
@@ -198,6 +202,7 @@ window.addEventListener('load', function () {
     // Attach click events to the main menu buttons and poller div
     document.querySelector('div#virtualserversbutton').addEventListener('click', showVirtualServers);
     document.querySelector('div#poolsbutton').addEventListener('click', showPools);
+	document.querySelector('div#policybutton').addEventListener('click', showPolicy);
```
Comment (epacke) [inline underlay/js/bigipreport.js:205]: Don't use tab for whitespace. Use spaces.

File: underlay/js/bigipreport.js [hunk 3]
```diff
@@ -1061,6 +1086,9 @@ function populateSearchParameters(updatehash) {
                 case 'irules':
                     showiRules(updatehash);
                     break;
+		    case 'policy':
```
Comment (epacke) [inline underlay/js/bigipreport.js:1089]: Replace tab here too. You can replace using regex by replacing `\t` with `    ` (space x4).

Comment (epacke) [reply underlay/js/bigipreport.js:1089]: I see a few more but won't comment on this again to keep the review neat and tidy.

**Review by epacke — CHANGES_REQUESTED** [review]

**Comment by epacke** [comment]
Almost there Marius! :)

**Pull request merged**